lib/logstash/devutils/rspec/shared_examples.rb

```diff
@@ -0,0 +1,19 @@
+require 'rspec/wait'
+
+RSpec.shared_examples "an interruptible input plugin" do
+  describe "stop" do
+    let(:queue) { SizedQueue.new(20) }
+    subject do
+      s = described_class.new(config)
+      s.register
+      s
+    end
+    it "returns from run" do
+      thread = Thread.new(subject, queue) { |subject, queue| subject.run(queue) }
+      consumer_thread = Thread.new(queue) { |queue| loop { queue.pop } }
+      wait(1).for { thread }.to be_alive
```

Comment:
what is this sorcery? I love it. `rspec/wait` from https://github.com/laserlemon/rspec-wait ?